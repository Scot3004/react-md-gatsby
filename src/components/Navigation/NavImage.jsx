import React, { Component } from "react";
import {CardTitle} from "react-md";
import "./NavImage.scss";

class NavImage extends Component {
  render() {
    const { config } = this.props;
    return (
      <div className="nav-image" >
        <CardTitle
          avatar={<img
            role="presentation"
            src={config.userAvatar}
            alt={config.userName}
            style={{
              width: 72,
              height: 72,
              padding: 10
            }}
          />}
          title={config.userName}
          subtitle="Web developer"
          style={{color: "white"}}
        />
      </div>
    )
  }
}

export default NavImage;
