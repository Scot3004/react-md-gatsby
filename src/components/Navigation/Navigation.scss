.main-content {
  display: flex;
  flex-direction: column;

  .main-container {
    display: flex;
    min-height: 100%;
    min-height: calc(100vh - 168px);
    flex-direction: column;
    flex: 1;

    .main-content {
      flex: 1;
    }
  }
}
