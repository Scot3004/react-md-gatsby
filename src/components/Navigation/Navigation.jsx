import React, { Component } from "react";
import NavigationDrawer from "react-md/lib/NavigationDrawers";
import ToolbarActions from "../ToolbarActions/ToolbarActions";
import Footer from "../Footer/Footer";
import GetNavList from "./NavList";
import NavImage from "./NavImage";
import "./Navigation.scss";

class Navigation extends Component {
  render() {
    const { children, config, LocalTitle } = this.props;
    const footerLinks = LocalTitle !== "About";
    return (
      <NavigationDrawer
        drawerTitle={<NavImage config={config} />}
        toolbarTitle={LocalTitle}
        contentClassName="main-content"
        navItems={GetNavList(config)}
        mobileDrawerType={NavigationDrawer.DrawerTypes.TEMPORARY}
        tabletDrawerType={NavigationDrawer.DrawerTypes.TEMPORARY}
        desktopDrawerType={NavigationDrawer.DrawerTypes.PERMANENT_MINI}
        toolbarActions={<ToolbarActions config={config} />}
      >
        <div className="main-container">{children}</div>
        <Footer userLinks={footerLinks} />
      </NavigationDrawer>
    );
  }
}

export default Navigation;
