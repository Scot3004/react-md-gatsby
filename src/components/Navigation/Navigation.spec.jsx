import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Navigation from './Navigation';


it('renders correctly', () => {
  const config = {
    userAvatar: "userAvatar",
    userName: "userName",
  }
  const tree = shallow(<Navigation config={config} />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('renders social links', () => {
  const config = {
    userAvatar: "userAvatar",
    userName: "userName",
    userLinks: [
      {
        label: "LinkLabel",
        url: "https://example.com",
        iconClassName: "fa fa-github"
      }
    ]
  }
  const tree = shallow(<Navigation config={config} />);
  expect(toJson(tree)).toMatchSnapshot();
});
