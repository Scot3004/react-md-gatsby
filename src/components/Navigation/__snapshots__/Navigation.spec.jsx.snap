// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<NavigationDrawer
  autoclose={true}
  constantDrawerType={false}
  contentClassName="main-content"
  contentComponent="main"
  contentId="main-content"
  defaultMedia="mobile"
  desktopDrawerType="full-height"
  desktopMinWidth={1025}
  drawerTitle={
    <NavImage
      config={
        Object {
          "userAvatar": "userAvatar",
          "userName": "userName",
        }
      }
    />
  }
  drawerTransitionDuration={300}
  extractMini={true}
  includeDrawerHeader={true}
  jumpLabel="Jump to content"
  mobileDrawerType="temporary"
  mobileMinWidth={320}
  navItems={
    Array [
      Object {
        "component": [Function],
        "leftIcon": <FontIcon
          iconClassName="material-icons"
        >
          home
        </FontIcon>,
        "primaryText": "Home",
        "to": "/",
      },
      Object {
        "divider": true,
      },
      Object {
        "divider": true,
      },
      Object {
        "component": [Function],
        "leftIcon": <FontIcon
          iconClassName="material-icons"
        >
          person
        </FontIcon>,
        "primaryText": "About",
        "to": "/about/",
      },
    ]
  }
  persistentIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      arrow_back
    </FontIcon>
  }
  position="left"
  tabletDrawerType="temporary"
  tabletMinWidth={768}
  temporaryIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      menu
    </FontIcon>
  }
  toolbarActions={
    <Toolbar
      config={
        Object {
          "userAvatar": "userAvatar",
          "userName": "userName",
        }
      }
    />
  }
  toolbarThemeType="colored"
  transitionEnterTimeout={300}
  transitionName="md-cross-fade"
>
  <div
    className="main-container"
  />
  <Footer
    userLinks={true}
  />
</NavigationDrawer>
`;

exports[`renders social links 1`] = `
<NavigationDrawer
  autoclose={true}
  constantDrawerType={false}
  contentClassName="main-content"
  contentComponent="main"
  contentId="main-content"
  defaultMedia="mobile"
  desktopDrawerType="full-height"
  desktopMinWidth={1025}
  drawerTitle={
    <NavImage
      config={
        Object {
          "userAvatar": "userAvatar",
          "userLinks": Array [
            Object {
              "iconClassName": "fa fa-github",
              "label": "LinkLabel",
              "url": "https://example.com",
            },
          ],
          "userName": "userName",
        }
      }
    />
  }
  drawerTransitionDuration={300}
  extractMini={true}
  includeDrawerHeader={true}
  jumpLabel="Jump to content"
  mobileDrawerType="temporary"
  mobileMinWidth={320}
  navItems={
    Array [
      Object {
        "component": [Function],
        "leftIcon": <FontIcon
          iconClassName="material-icons"
        >
          home
        </FontIcon>,
        "primaryText": "Home",
        "to": "/",
      },
      Object {
        "divider": true,
      },
      Object {
        "component": "a",
        "href": "https://example.com",
        "leftIcon": <FontIcon
          forceSize={true}
          iconClassName="fa fa-github"
        />,
        "primaryText": "LinkLabel",
      },
      Object {
        "divider": true,
      },
      Object {
        "component": [Function],
        "leftIcon": <FontIcon
          iconClassName="material-icons"
        >
          person
        </FontIcon>,
        "primaryText": "About",
        "to": "/about/",
      },
    ]
  }
  persistentIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      arrow_back
    </FontIcon>
  }
  position="left"
  tabletDrawerType="temporary"
  tabletMinWidth={768}
  temporaryIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      menu
    </FontIcon>
  }
  toolbarActions={
    <Toolbar
      config={
        Object {
          "userAvatar": "userAvatar",
          "userLinks": Array [
            Object {
              "iconClassName": "fa fa-github",
              "label": "LinkLabel",
              "url": "https://example.com",
            },
          ],
          "userName": "userName",
        }
      }
    />
  }
  toolbarThemeType="colored"
  transitionEnterTimeout={300}
  transitionName="md-cross-fade"
>
  <div
    className="main-container"
  />
  <Footer
    userLinks={true}
  />
</NavigationDrawer>
`;
